### ISSUE TITLE
toArray fix in #80 causes other issues

### ISSUE THREAD

**Issue by xHeaven** [WendellAdriel/laravel-validated-dto#89]
We just realized at the team that v3.6.1 (https://github.com/WendellAdriel/laravel-validated-dto/pull/80) just broke something pretty important (for us, that is).

Lets pretend we have this DTO:
```php
class TestDTO extends SimpleDTO
{
    #[Map(data: 'data')]
    public array $foobar;

    protected function defaults(): array
    {
        return [];
    }

    protected function casts(): array
    {
        return [];
    }
}
```

Lets use this DTO like this:
```php
TestDTO::fromArray(['data' => ['12345']]);
```

In `3.6.0`, this code outputs this:
```
App\TestDTO {
    +foobar: [
      "12345",
    ],
}
```

However, after `3.6.1`, the output is this:
```
App\TestDTO {
    +foobar: [],
}
```

This is a quite breaking change, as we can't really tell random API providers to not provide a key called `data`. We also wouldn't like to manually map the `data` key into something else, I mean that's the entire point of having a DTO that does this for us in the first place.

Can we get this fixed, or is there a workaround that doesn't involve manual mapping?

Thanks!

**Comment by WendellAdriel** [comment]
Sorry for the delay, I had a lot going on.

Oh, so you were using the `data` as a mapped property?
Oh man, since I was using this for internal things only I thought it wouldn't break anything.
I'll test some things out and release a new version ASAP, thanks for letting me know and sorry for this.

**Comment by WendellAdriel** [comment]
This was fixed in the v3.9.0
